### PR TITLE
chore: Enzyme replacement in test files- PR 2 of 6 for part 2 of 3

### PR DESCRIPTION
**Pull request by rohittarpara**
#### Details

Enzyme replacement in test files.

##### Motivation

enzyme replacement in files under /src/tests/unit/tests/DetailsView/components/

##### Context

All changes are done in test files only and snapshots are updated.
details-view-command-bar.test.tsx: We will raise this file in part 3 feature PRs.

#### Pull request checklist

- [x] Addresses an existing issue: [USER STORY 2136483](https://dev.azure.com/mseng/1ES/_workitems/edit/2136483)
- [x] Ran `yarn fastpass`
- [x] Added/updated relevant unit test(s) (and ran `yarn test`)
- [x] Verified code coverage for the changes made. Check coverage report at: `<rootDir>/test-results/unit/coverage`
- [x] PR title *AND* final merge commit title both start with a semantic tag (`fix:`, `chore:`, `feat(feature-name):`, `refactor:`). See `CONTRIBUTING.md`.
- [N/A] (UI changes only) Added screenshots/GIFs to description above
- [N/A] (UI changes only) Verified usability with NVDA/JAWS

### CODE REVIEW

**Comment by DaveTryon** [comment]
General comment: Having a series of PR's titled "... user story ABC", "... user story DEF", "...user story GHI", etc., can be useful when tracking the implementation of the work, but once the PR is merged, its value drops dramatically, especially if the engineering system ever becomes unavailable. As an alternative, may I suggest a slightly different approach, with the intent to make it possible to understand the change even without access to the ADO system (which will be the case for any external contributor, or even internal contributors should we later change engineering systems)

- The title should capture the fact that this is part of a broader set of tasks. All tasks in the set will ideally use the same pattern
- The description should still point to the ADO item, but will also include the set of work included in this task. There will be some duplication with ADO--that's a good thing.

So a specific one of these tasks might have a title of "chore: Enzyme replacement in test files - part 3 of 12". The description might include something like:

Files in scope for this PR:
- src/tests/unit/tests/DetailsView/components/awesome-class.test.tsx
- src/tests/unit/tests/DetailsView/components/amazing-component.test.tsx
- src/tests/unit/tests/DetailsView/components/incredible-navigation.test.tsx
  etc.

File: src/tests/unit/tests/DetailsView/components/generic-dialog.test.tsx [hunk 1]
```diff
@@ -69,9 +89,15 @@ describe('GenericDialogTest', () => {
...
-        rendered.find(Dialog).prop('onDismiss')();
+        useOriginalReactElements('@fluentui/react', [
+            'Dialog',
+            'DialogFooter',
+            'PrimaryButton',
+            'DefaultButton',
+        ]);
+        const renderResult = render(<GenericDialog {...props} />);
+        const button = renderResult.getByRole('button', { name: /Cancel/i });
+        fireEvent.click(button);
```
Comment (madalynrose) [inline src/tests/unit/tests/DetailsView/components/generic-dialog.test.tsx:100]: We can keep this test mostly the same as the original test
```suggestion
        render(<GenericDialog {...props} />);
        getMockComponentClassPropsForCall(Dialog).onDismiss();
```

**Review by madalynrose — CHANGES_REQUESTED** [review]
A few small suggestions and then this is good to go!

File: src/tests/unit/tests/DetailsView/components/export-dropdown.test.tsx
[the remark this answers is not in the document]
Comment (rohittarpara) [reply src/tests/unit/tests/DetailsView/components/export-dropdown.test.tsx:1]: Done, please review on latest commit

File: src/tests/unit/tests/DetailsView/components/export-dialog.test.tsx [hunk 2]
```diff
@@ -1,18 +1,39 @@
...
 describe('ExportDialog', () => {
+    mockReactComponents([
+        Dialog,
+        PrimaryButton,
+        TextField,
+        DialogFooter,
+        CodePenReportExportService.exportForm,
+        ExportDropdown,
+    ]);
+    //CodePenReportExportService.exportForm;
```
[the remark this answers is not in the document]
Comment (rohittarpara) [reply src/tests/unit/tests/DetailsView/components/export-dialog.test.tsx]: Done, please review on latest commit

File: src/tests/unit/tests/DetailsView/components/generic-dialog.test.tsx (hunk 1 above)
Comment (rohittarpara) [reply src/tests/unit/tests/DetailsView/components/generic-dialog.test.tsx]: Done, please review on latest commit

**Review by madalynrose — APPROVED** [review]

**Pull request merged**